client/src/details/SFSClientImpl.cpp

```diff
@@ -391,7 +422,8 @@ std::string SFSClientImpl<ConnectionManagerT>::GetBaseUrl() const
     {
         return *m_customBaseUrl;
     }
-    return "https://" + m_accountId + "." + std::string(c_apiDomain);
+
+    return "https://" + SFSUrlComponents::UrlEscape(m_accountId, m_reportingHandler) + "." + std::string(c_apiDomain);

```

Comment:
why not escape the entire string?  Currently c_apiDomain  might not need escaping, but this will break if that constant changes and someone doesnt change this line.

Reply:
Other characters also get escaped, like : and /
If someone changes c_apiDomain they will know as it will break immediately.

Reply:
let me clarify: why not escape the entire string aside from the `scheme://`?  If a character should be escaped, UrlEscape will escape it.